REST/PowerShell/Deployments/GetLatestDeploymentForAllProjects.ps1

```diff
@@ -7,56 +7,94 @@ $headers = @{ "X-Octopus-ApiKey" = $apiKey }
 # Get space ID
 $space = (Invoke-RestMethod -Method Get -Uri "$octopusURL/api/spaces/all" -Headers $headers) | Where-Object { $_.Name -eq $spaceName }
 $spaceId = $space.Id
+$octopusSpaceUrl = "$octopusURL/api/$spaceId"
 
-# Get all projects in the specified space
-$projectListUrl = "$octopusURL/api/$spaceId/projects/all"
-$projects = Invoke-RestMethod -Uri $projectListUrl -Method Get -Headers $headers
+$showProgress = $true
+
+# Get first page of projects
+$take = 30
+$count = 1
+$pageNumber = 0
+Write-Host "Getting page 1 of projects for space $spaceId"
+$projects = Invoke-RestMethod -Uri "$octopusSpaceUrl/projects?take=$take" -Headers $headers
+$total = $projects.TotalResults
 
 # Initialize an array to hold the project details
 $projectDetails = @()
 
 Write-Host "---"
-Write-Host "Calculating the latest deployment date for $($projects.count) projects (this may take some time!)."
+Write-Host "Calculating the latest deployment date for $total projects (this may take some time!)."
 Write-Host "---"
 
-foreach ($project in $projects) 
-{
-    $deploymentsUrl = "$octopusURL/api/$spaceId/deployments?projects=$($project.Id)&take=1"
-    try {
-        $latestDeployment = Invoke-RestMethod -Uri $deploymentsUrl -Method Get -Headers $headers -ErrorAction Stop | Select-Object -ExpandProperty Items | Select-Object -First 1
+while ($pageNumber -le $projects.LastPageNumber) {
+    foreach ($project in $projects.Items) {
+        $per = [math]::Round($count / $total * 100.0, 2) - 1

```

Comment:
The only concern might be if there are no results, $total will be 0, and you could end up with a divide-by-zero issue here.